netlify/cli
# feat: read scopes and contexts in `env:get`, `env:list`, and `dev`

#### Summary
This PR adds support for environment variable scopes and contexts for the `env:get`, `env:list`, and `dev` commands. 

`env:get` and `env:list` support both the `--scope` and `--context` flags if the site has been configured with the new Environment Variables experience:tm:, aka Envelope. If the site is not configured with Envelope and either of those flags are passed, an error message will print. If this site is not configured with Envelope and neither of those flags are passed, it will behave as it normally has. 

A new Scope column has been added to `env:list` if the site is on Envelope: 
<img width="722" alt="Screenshot 2022-07-27 at 22 29 27" src="https://user-images.githubusercontent.com/236451/181427814-2b832ef6-21d8-4234-89e0-a358043312bc.png">

The `--context` flag for `netlify dev` is now supported, which will pull in values from Envelope while respecting the precedence of context values in netlify.toml. 

One caveat is that there will be a duplicate GET request sent to Envelope due to [this change in netlify/config](https://github.com/netlify/build/pull/4329). Since we're adding context and scope options on the CLI level, I thought it would make sense to move the requests to happen in the CLI. Once all the calls to Envelope land in CLI, it should be safe to remove in netlify/config to remove the redundancy. 

Fixes https://github.com/netlify/pillar-workflow/issues/755
Fixes https://github.com/netlify/pillar-workflow/issues/734

---

## Test instructions

- Checkout this branch and install dependencies
- Remove your existing global install -- if installed with npm, `npm uninstall -g netlify-cli`
- `npm link`
- Give `env:get`, `env:list`, and `dev` a whirl. I recommend trying with a site opted-in to the new Env Vars experience in the UI (enable in Labs) 

---

For us to review and ship your PR efficiently, please perform the following steps:

- [x] Read the [contribution guidelines](../CONTRIBUTING.md) 📖. This ensures your code follows our style guide and
      passes our tests.

**A picture of a cute animal (not mandatory, but encouraged)**

<img width="485" alt="Screenshot 2022-07-24 at 21 41 04" src="https://user-images.githubusercontent.com/236451/180700011-574a34cb-d777-4d49-9cd3-65f8c51ac181.png">
